src/main/java/it/gov/pagopa/pu/sil/service/paymentnotification/PaymentNotificationService.java

```diff
@@ -42,13 +52,24 @@ public void notifyPayment(Long orgSilServiceId, InstallmentDTO installmentDTO, U
       .orElseThrow(() -> new IllegalArgumentException("Organization having id " + orgSilService.getOrganizationId() + " not found"));
 
     DebtPosition debtPosition = debtPositionService.getDebtPositionByInstallmentId(installmentDTO.getInstallmentId(), accessToken);
-    if(debtPosition == null){
+    if (debtPosition == null) {
       throw new IllegalArgumentException("DebtPosition related to installmentId " + installmentDTO.getInstallmentId() + " not found");
     }
-    if(!debtPosition.getOrganizationId().equals(organization.getOrganizationId())){
+    if (!debtPosition.getOrganizationId().equals(organization.getOrganizationId())) {
       throw new IllegalArgumentException("The installment provided is not related to the organization requested. requested " + organization.getOrganizationId() + " provided " + debtPosition.getOrganizationId());
     }
 
+    if (BooleanUtils.isTrue(orgSilService.getFlagLegacy())) {
+      //legacy payment notification implementation
+      legacyPaymentNotification(installmentDTO, organization, orgSilService, loggedUser, accessToken);
+    } else {
+      //native payment notification implementation
+      nativePaymentNotification(installmentDTO, organization, orgSilService, loggedUser, accessToken);
```

Comment:
let's avoid to use the `native` prefix... it's implicit
```suggestion
      paymentNotification(installmentDTO, organization, orgSilService, loggedUser, accessToken);
```